orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/tasks/BatchUpdatePipelinesTask.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2019 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.orca.front50.tasks;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.netflix.spinnaker.orca.ExecutionStatus;
+import com.netflix.spinnaker.orca.RetryableTask;
+import com.netflix.spinnaker.orca.TaskResult;
+import com.netflix.spinnaker.orca.front50.Front50Service;
+import com.netflix.spinnaker.orca.front50.PipelineModelMutator;
+import com.netflix.spinnaker.orca.pipeline.model.Stage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import retrofit.client.Response;
+
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+@Component
+public class BatchUpdatePipelinesTask implements RetryableTask {
+
+  private Logger log = LoggerFactory.getLogger(getClass());
+
+  @Autowired(required = false)
+  private Front50Service front50Service;
+
+  @Autowired(required = false)
+  private List<PipelineModelMutator> pipelineModelMutators = new ArrayList<>();
+
+  @Autowired
+  ObjectMapper objectMapper;
+
+  @SuppressWarnings("unchecked")
+  @Override
+  public TaskResult execute(Stage stage) {
+    if (front50Service == null) {
+      throw new UnsupportedOperationException("Front50 is not enabled, no way to save pipeline. Fix this by setting front50.enabled: true");
+    }
+
+    if (!stage.getContext().containsKey("pipelines")) {
+      throw new IllegalArgumentException("pipeline contexts must be provided");
+    }
+
+    if (!(stage.getContext().get("pipelines") instanceof String)) {
+      throw new IllegalArgumentException("'pipelines' context key must be a base64-encoded string: Ensure you're on the most recent version of gate");
+    }
+
+    byte[] pipelineData;
+    try {
+      pipelineData = Base64.getDecoder().decode((String) stage.getContext().get("pipelines"));
```

Comment:
Maybe I am missing something, but seems a bit wasteful to run the base64 decoder twice (here and in `Stage.decodeBase64` below). Doesn't seem like it adds that much value in terms of better error messaging. Could just catch on the line 76

Reply:
Thanks, had just copied this from the `SavePipelineTask` but agree will be simpler to just decode and catch once -- updated!